lib/automation/llm_triage.rb

```diff
@@ -82,6 +83,25 @@ def self.handle(
               force_review: true,
             )
           end
+
+          # note this does not notify user of the action
+          # as it is mainly used for spam where notification
+          # is not needed.
+          # we will need another flag if we want to notify
+          if hide_post
```

Comment:
I feel like you could lean on `post.hide!` in core here https://github.com/discourse/discourse/blob/c446caa6e9f68be24bdc3d0ff0b33e0ddc2130c6/app/models/post.rb#L604-L604 since it's doing the same thing the code does here, and give it `PostActionType.types[:spam]` as the `post_action_type_id`. I think the only thing you would need to do is your additional topic visibility update.

Reply:
Also generally I am not sure this with the `flag_post/needs_review` part above is quite the correct way to go, seems like going through the `PostActionCreator` and indicating that it's for spam :

```ruby
PostActionCreator.new(
  Discourse.system_user,
  post,
  PostActionType.types[:spam],
  message: message,
  queue_for_review: true,
).perform
```

This auto-hides if necessary _and_ also does the `needs_review!` on the flagged post. If we did this the `hide_topic` and `hide_post` options might not be necessary for the automation.

It feels like we are not relying on the review queue logic to properly mark this as spam, and since this is the case if you mark the reviewable invalid / disagree the post might not get unhidden as you already indicated, and there is no testing for this.

The review queue logic is super hard to follow though so I understand the appeal of doing it in a more basic way here, I just think it may cause other problems doing it this way. This may solve the customer's  immediate problem but I am not sure it's maintanable. Maybe we can do some further digging first and try out `PostActionCreator` before merging this?

Reply:
Might also be worth having 2 options:

1. Send to review - Leave this as is, just ask for post approval
2. Mark as spam - Follow the spam flow with PostActionCreator, which would also do the hiding

Reply:
its tricky cause hide_topic is used for automations that do not flag, we can not get rid of it. But I can try to see if I can get PostActionCreator going and allow people to select "send to review/mark as spam/don't involve review" ... its so complicated :(

Reply:
Worth mentioning that back on the old toxicity / nsfw days we used PostActionCreator and had problems because people don't always want the users to be aware that they were flagged / auto-hidden, which happens on both public topics and PMs.

This is the main reason we now only send to the queue without all the usual flag adjacent side effects. But that also means that if people now want it, it needs to be an option much like what @martin-brennan described.